src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -178,6 +179,16 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
             {
                 try
                 {
+                    // FeatureManagement assemblies may not be loaded on provider startup, so version information is gathered upon first refresh
+                    if (!_featureManagementVersionsTelemetrySet)
+                    {
+                        _requestTracingOptions.FeatureManagementVersion = TracingUtils.GetAssemblyVersion(RequestTracingConstants.FeatureManagementAssemblyName);
+
+                        _requestTracingOptions.FeatureManagementAspNetCoreVersion = TracingUtils.GetAssemblyVersion(RequestTracingConstants.FeatureManagementAspNetCoreAssemblyName);
+
+                        _featureManagementVersionsTelemetrySet = true;
```

Comment:
I don't think this is something we should attempt repeatedly. I would move this line before the GetAssemblyVersion call... but I'm still concerned this may skip the first refresh.

Reply:
Why would this be attempted repeatedly? From what I understand, `GetAssemblyVersion` wouldn't prevent the code from reaching the line where we set the boolean flag.

Reply:
It will never throw?

Reply:
It can throw some rare exceptions like `AppDomainUnloadedException` but I think it makes sense to move the flag setting to the top anyway now